src/CommandingOfficers/AW4/IDS/Penny.java

```diff
@@ -83,11 +87,17 @@ private static class Stormfront extends RuinedAbility
     @Override
     public GameEventQueue getEvents(MapMaster map)
     {
-      Weathers[] candidates = { Weathers.SLEET, Weathers.SMOKE, Weathers.SIROCCO };
-      int rand = map.game.getRN(candidates.length);
+      ArrayList<Weathers> candidates = new ArrayList<>();
+      candidates.add(Weathers.SLEET);
+      candidates.add(Weathers.SMOKE);
+      candidates.add(Weathers.SIROCCO);
+      for( Unit u : COcast.COUs ) // Don't roll the same weather that's currently active
+        candidates.remove(map.getEnvironment(u.x, u.y).weatherType);
```

Comment:
And what if Penny has a COU in each of the three weather types? Wait, that can't happen? We should give Penny multiple COUs, then.

Reply:
If you wanna be really pedantic about it, I guess I could check for the weather the *unit who activated the power* is in.